components/ShowDirection.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import { StyleSheet, Image } from 'react-native';
+import Polyline from '@mapbox/polyline';
+import MapView, { Marker } from 'react-native-maps';
+import Colors from '../constants/Colors';
+import transportMode from '../classes/transportMode';
+
+const styles = StyleSheet.create({
+  imageSize: {
+    width: 30,
+    height: 30,
+  },
+});
+
+type directionProps = {
+  startLat: number;
+  startLon: number;
+  endLat: number;
+  endLon: number;
+  transportType: transportMode;
+};
+
+class ShowDirection extends Component<directionProps, { steps: any[] }> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      steps: [],
+    };
+    this.getDirectionsSteps = this.getDirectionsSteps.bind(this);
+  }
+
+  componentDidMount() {
+    this.getDirectionsSteps();
+  }
+
+  // This function takes adresses as input and call the google API.
+  // Time and distance are unused but left here for potential extra.
+  async getDirectionsSteps() {
+    const { startLat, startLon, endLat, endLon, transportType } = this.props;
+    const url = `https://maps.googleapis.com/maps/api/directions/json?origin=${startLat},${startLon}&destination=${endLat},${endLon}&mode=${transportType}&key=AIzaSyDpY_ACPWoo3mVPPmiLKQe1aBhjkDYlwJI`;
```

Comment:
This line is pretty big. Is there a way to make it shorter? The linter did not trigger any warning on my IDE, I assume for react native standards, it okay, but for readability it is not the best.